### PR TITLE
Merge master into refactor redux part i

### PR DESCRIPTION
**Pull request by bartaz**

### CODE REVIEW

File: src/common/reducers/entities/index.js [hunk 1]
```diff
@@ -12,6 +12,11 @@ export function entities(state = { snaps: {}, repos: {} }, action) {
     return merge({}, state, action.payload.entities);
   }
 
+  // TODO merge master - duplication of above, do we need to support both formats?
+  if (action.payload && action.payload.response && action.payload.response.entities) {
```
Comment (bartaz) [inline src/common/reducers/entities/index.js]: One of the issues after merge is that `entities` from API may be directly in action `payload` (if fetch was done directly) or in `response` prop (if it went through CALL_API middleware).

File: src/common/reducers/entities/index.js [hunk 2]
```diff
@@ -12,7 +12,9 @@ export function entities(state = { snaps: {}, repos: {} }, action) {
     return merge({}, state, action.payload.entities);
   }
 
-  // TODO merge master - duplication of above, do we need to support both formats?
+  // XXX
+  // some action that don't use CALL_API middleware pass entities in payload directly
+  // while CALL_API middleware does it in response prop
   if (action.payload && action.payload.response && action.payload.response.entities) {
```
Comment (bartaz) [inline src/common/reducers/entities/index.js]: We need to read `entities` both from action payload directly and response object to support current state of 2 ways of calling APIs (direct or via CALL_API).

File: src/common/reducers/snaps.js [hunk 3]
```diff
@@ -51,17 +51,17 @@ export function snaps(state = {
         isFetching: false,
         success: true,
         snaps: [
-          ...action.payload.snaps
+          ...action.payload.response.payload.snaps
```
Comment (bartaz) [inline src/common/reducers/snaps.js]: Nested in a bit ugly way, but let's look into that later when we refactor snaps store.

File: src/common/reducers/entities/index.js [hunk 4]
```diff
@@ -12,6 +12,13 @@ export function entities(state = { snaps: {}, repos: {} }, action) {
     return merge({}, state, action.payload.entities);
   }
 
+  // XXX
+  // some action that don't use CALL_API middleware pass entities in payload directly
```
Comment (earnubs) [inline src/common/reducers/entities/index.js]: ref #465

File: src/common/store/configureStore.js [hunk 5]
```diff
@@ -2,18 +2,25 @@ import { createStore, applyMiddleware, compose } from 'redux';
...
 import { browserHistory } from 'react-router';
 import thunk from 'redux-thunk';
+
 import rootReducer from '../reducers';
 import { analytics } from '../middleware';
+import callApi from '../middleware/call-api';
+import { conf } from '../helpers/config';
 
-export default function configureStore(preloadedState) {
+export default function configureStore(preloadedState, csrfToken) {
```
Comment (earnubs) [inline src/common/store/configureStore.js]: it's just occurred to me that the token could have been in preloadedstate

**Pull request merged**